### ISSUE TITLE
Misaligned column headers in the main demo

### ISSUE THREAD

**Issue by krzysztofspilka** [handsontable/handsontable#9580]
### Description

Open the dropdown menu in the second column “Name”. Keep it open, scroll to the last column. Click on any cell in the last column (Country), then any cell in the Order ID column.

Result: misaligned column headers

https://user-images.githubusercontent.com/8048526/173592507-ef1758d8-d594-4822-83a9-f8fde7a5e276.mov

### Links

https://handsontable.com/docs/demo/

**Comment by warpech** [comment]
AFAIK this is only reproducible in an `iframe`.

**Comment by wszymanski** [comment]
As @warpech wrote, this bug is only replicable in `iframe`. The bug is related to the fact that drop-down menu (or context menu) has greater height than `iframe`.

<img width="925" alt="Screenshot 2022-09-16 at 16 56 53" src="https://user-images.githubusercontent.com/141330/190669368-d35f5667-5973-4857-9300-3ac7f10b9696.png">

After scrolling right after opening the menu, you can see that there is an extra scroll.

<img width="938" alt="Screenshot 2022-09-16 at 16 57 11" src="https://user-images.githubusercontent.com/141330/190669977-b2b75dbb-c956-4ab7-8a45-9a9ac6f52e99.png">

I simplified the demo, as it is quite complex. I can't find a way for reproducing the bug on one page (without `iframe`). 
I tested whether an extra `div` (see below) can trim the context menu (the same as makes `iframe`).

`Code`

```html
<div style="width: 990px; height: 466px; overflow: hidden;">
  <div id="example1" class="hot "></div>
</div>
```

`Result`

<img width="1011" alt="Screenshot 2022-09-16 at 17 04 37" src="https://user-images.githubusercontent.com/141330/190673549-925ce618-2464-4a2b-8851-fa70a7b5867c.png">

It looks looks different than in the `body` when we have an `iframe` (see below).

<img width="1016" alt="Screenshot 2022-09-16 at 17 09 56" src="https://user-images.githubusercontent.com/141330/190674510-aec250bc-e373-4352-b0b2-a444fcb29a32.png">

I've found that [#6202 ](https://github.com/handsontable/handsontable/issues/6202) issue addressed problem of cutting floating elements by the inner window bounding box. I tested whether `uiContainer` option introduced within https://github.com/handsontable/handsontable/pull/6304 give an ability to attach floating element to a "main" window (using `window.frameElement` method). I was not successful. Anyway, I abandoned further attempts, coming to the conclusion that there is no point in complicating the example.

After concluding that there is need a fix, without changing the configuration I found that there are some cases which may be related.

1. The main problem, shown in a [simplified example](https://gist.github.com/wszymanski/3fe3490a8a912d3cdde0c7d6e7c5da71).

https://user-images.githubusercontent.com/141330/190698725-6db77412-9f2f-473f-ac15-f291283e2f52.mp4

2. Wrong with of column headers when option `rowHeaders` is set to `false` while scrolling (the `ContextMenu` plugin).

https://user-images.githubusercontent.com/141330/190696742-e9cf802b-c51b-4aeb-83e1-0cc51b311697.mp4

3. Wrong with of column headers when option `rowHeaders` is set to `false` while scrolling (the `DropdownMenu` plugin).

https://user-images.githubusercontent.com/141330/190699138-34ee6df7-ded6-483a-adab-6814dc198d1d.mp4

I've found that problem is not related to `AutoColumnSize` and `AutoRowSize` plugins. Disabling them won't change anything. The problem is related to part of code which calculate scrollbars (see below).

https://github.com/handsontable/handsontable/blob/08895d41f91294e5de8521b05423c951a0f8cf8d/handsontable/src/3rdparty/walkontable/src/overlay/top.js#L188-L202

I think that there should be found proper way of calculating scrollbars width, but so far I haven't found working solution.

**Comment by wszymanski** [comment]
As discussed with @warpech and @budnix I'm changing the assignee.

**Comment by aninde** [comment]
Fixed by v12.3.
